integration_tests/src/test/scala/com/nvidia/spark/rapids/tests/tpch/TpchLikeSparkSuite.scala

```diff
@@ -91,113 +98,124 @@ class TpchLikeSparkSuite extends FunSuite with BeforeAndAfterAll {
     }).max())
   }
 
+  private def assertResult(expectedRowCount: Int)(df: DataFrame): Unit = {
+    ExecutionPlanCaptureCallback.startCapture()
+    val c = df.count()
+    val plan = ExecutionPlanCaptureCallback.getResultWithTimeout()
```

Comment:
@revans2 raised concern earlier about a single SQL query executing multiple jobs (e.g.: building broadcast table separately from join on broadcast table).  Is that still a concern here?

Reply:
It isn't something I have seen happen yet but I think we could fairly easily modify the `ExecutionPlanCaptureCallback` to store multiple plans if/when we need this functionality. Tests could make multiple calls to `getResultWithTimeout` in cases where multiple plans are expected. Let's see what @revans2 says though.